core/payment-driver/base/src/driver.rs

```diff
@@ -88,4 +83,27 @@ pub trait PaymentDriver {
         caller: String,
         msg: ValidateAllocation,
     ) -> Result<bool, GenericError>;
+
+    async fn sign_payment(
+        &self,
+        _db: DbExecutor,
+        _caller: String,
+        msg: SignPayment,
+    ) -> Result<Vec<u8>, GenericError> {
+        let payload = utils::payment_hash(&msg.0);
+        let node_id = msg.0.payer_id;
+        bus::sign(node_id, payload).await
+    }
+
+    async fn verify_signature(
+        &self,
+        _db: DbExecutor,
+        _caller: String,
+        msg: VerifySignature,
+    ) -> Result<bool, GenericError> {
+        let payload = utils::payment_hash(&msg.payment);
+        let node_id = msg.payment.payer_id;
+        let signature = bus::sign(node_id, payload).await?;
```

Comment:
isnt `verify_signature` executed on provider where the `payer_id` private key is not available?

Reply:
if above thought is correct you can use something like: https://github.com/tomusdrw/ethsign/blob/9ff18e158dda29b13558c04f77441d93c7bbce57/src/key.rs#L83

Reply:
:facepalm: Obviously! I'll fix that.

Reply:
Fixed.